tests/components/shell_command/test_init.py

```diff
@@ -22,161 +20,151 @@ async def communicate() -> Tuple[bytes, bytes]:
         """
         return b"I am stdout", b"I am stderr"
 
-    mock_process = Mock()
+    mock_process = MagicMock()
     mock_process.communicate = communicate
     mock_process.returncode = int(error)
     return mock_process
 
 
-class TestShellCommand(unittest.TestCase):
-    """Test the shell_command component."""
-
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started.
-
-        Also seems to require a child watcher attached to the loop when run
-        from pytest.
-        """
-        self.hass = get_test_home_assistant()
-        asyncio.get_child_watcher().attach_loop(self.hass.loop)
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_executing_service(self):
-        """Test if able to call a configured service."""
-        with tempfile.TemporaryDirectory() as tempdirname:
-            path = os.path.join(tempdirname, "called.txt")
-            assert setup_component(
-                self.hass,
-                shell_command.DOMAIN,
-                {shell_command.DOMAIN: {"test_service": f"date > {path}"}},
-            )
-
-            self.hass.services.call("shell_command", "test_service", blocking=True)
-            self.hass.block_till_done()
-            assert os.path.isfile(path)
-
-    def test_config_not_dict(self):
-        """Test that setup fails if config is not a dict."""
-        assert not setup_component(
-            self.hass,
+async def test_executing_service(hass):
+    """Test if able to call a configured service."""
+    with tempfile.TemporaryDirectory() as tempdirname:
+        path = os.path.join(tempdirname, "called.txt")
+        assert await async_setup_component(
+            hass,
             shell_command.DOMAIN,
-            {shell_command.DOMAIN: ["some", "weird", "list"]},
+            {shell_command.DOMAIN: {"test_service": f"date > {path}"}},
         )
+        await hass.async_block_till_done()
 
-    def test_config_not_valid_service_names(self):
-        """Test that setup fails if config contains invalid service names."""
-        assert not setup_component(
-            self.hass,
-            shell_command.DOMAIN,
-            {shell_command.DOMAIN: {"this is invalid because space": "touch bla.txt"}},
-        )
+        await hass.services.async_call("shell_command", "test_service", blocking=True)
+        await hass.async_block_till_done()
+        assert os.path.isfile(path)
 
-    @patch(
-        "homeassistant.components.shell_command.asyncio.subprocess"
-        ".create_subprocess_shell"
+
+async def test_config_not_dict(hass):
+    """Test that setup fails if config is not a dict."""
+    assert not await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)